azurerm/resource_arm_virtual_machine_scale_set.go

```diff
@@ -940,6 +947,14 @@ func flattenAzureRmVirtualMachineScaleSetNetworkProfile(profile *compute.Virtual
 					config["subnet_id"] = *properties.Subnet.ID
 				}
 
+				if properties.ApplicationGatewayBackendAddressPools != nil {
+					addressPools := make([]interface{}, 0, len(*properties.ApplicationGatewayBackendAddressPools))
```

Comment:
can we move this assignment out of the if statement and make this:

```
addressPools := make([]interface{}, 0)
if properties.ApplicationGatewayBackendAddressPools != nil {
  # ...
}
config["application_gateway_backend_address_pool_ids"] = schema.NewSet(schema.HashString, addressPools)
```

this will ensure the `application_gateway_backend_address_pool_ids` variable is set to an empty list if the field is nil (which will mean any changes are detected)

Reply:
Ack. Done.